src/wp-includes/post.php

```diff
@@ -2274,6 +2302,22 @@ function get_all_post_type_supports( $post_type ) {
 function post_type_supports( $post_type, $feature ) {
 	global $_wp_post_type_features;
 
+	if ( 'footnotes' === $feature ) {
+		/*
+		 * Footnotes require additional items to be supported.
+		 *
+		 * The post type must also support `editor`, `revisions` and `custom-fields`.
+		 */
+		$required_features = array( 'footnotes', 'editor', 'revisions', 'custom-fields' );
+		foreach ( $required_features as $required_feature ) {
+			if ( ! isset( $_wp_post_type_features[ $post_type ][ $required_feature ] ) ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
```

Comment:
Given the snowflake code in `add|remove_post_type_support` this block may no be needed. It accounts for developers modifying `$_wp_post_type_features` directly but it may be possible to consider doing so out of warranty.

If it's decided to keep this then similar code will probably be needed in `get_all_post_type_supports()` and `get_post_types_by_support()`.